modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/repository/cache/PortalCacheRegistryItemImpl.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.document.library.internal.repository.cache;
+
+import com.liferay.document.library.internal.repository.capabilities.LiferayDynamicCapability;
+import com.liferay.portal.kernel.cache.CacheRegistryItem;
+import com.liferay.portal.kernel.repository.DocumentRepository;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Renan Vasconcelos
+ */
+@Component(service = CacheRegistryItem.class)
+public class PortalCacheRegistryItemImpl implements CacheRegistryItem {
+
+	@Override
+	public String getRegistryName() {
+		Class<?> clazz = getClass();
+
+		return clazz.getName();
+	}
+
+	@Override
+	public void invalidate() {
+		_clearLiferayDynamicCapabilities();
+	}
+
+	private void _clearLiferayDynamicCapabilities() {
+		for (LiferayDynamicCapability liferayDynamicCapability :
+				_liferayDynamicCapabilities.values()) {
+
+			liferayDynamicCapability.clear();
+		}
+
+		_liferayDynamicCapabilities.clear();
+	}
+
+	private final Map<DocumentRepository, LiferayDynamicCapability>
```

Comment:
@RenanUmV @rafaprax After extracting this class, this map is never populated; on the other hand, the original map in `PortalCapabilityLocatorImpl` will not be cleared.

Reply:
@zhangyan0701 Do we have any test related to `PortalCapabilityLocator`?

Reply:
> @zhangyan0701 Do we have any test related to `PortalCapabilityLocator`?

Hi @dantewang What does `PortalCapabilityLocator` do? Are there manual steps that may trigger this class? I am not really familiar with DM repository.